tests/gracenote_tests.ts

```diff
@@ -106,7 +110,43 @@ function basic(options: TestOptions): void {
 
   ok(true, 'GraceNoteBasic');
 }
+function graceNoteModifiers(options: TestOptions): void {
+  const f = VexFlowTests.makeFactory(options, 700, 130);
+  const stave = f.Stave({ x: 10, y: 10, width: 650 });
+
+  const gracenotes = [{ keys: ['b/4'], duration: '8', slash: false }].map(f.GraceNote.bind(f));
+
+  const notes = [
+    f
+      .StaveNote({ keys: ['c/5'], duration: '4', auto_stem: true })
+      .addModifier(f.GraceNoteGroup({ notes: gracenotes }).beamNotes(), 0)
+      .addModifier(new Accidental('#')),
+    f
+      .StaveNote({ keys: ['c/5'], duration: '8', auto_stem: true })
+      .addModifier(f.GraceNoteGroup({ notes: gracenotes }).beamNotes(), 0),
+    f
+      .StaveNote({ keys: ['c/5'], duration: '4', auto_stem: true })
+      .addModifier(f.GraceNoteGroup({ notes: gracenotes }).beamNotes(), 0),
+    f
+      .StaveNote({ keys: ['c/5'], duration: '4', auto_stem: true })
+      .addModifier(f.GraceNoteGroup({ notes: gracenotes }).beamNotes(), 0),
+  ];
+  notes[0].addModifier(new Articulation('a-').setPosition(3), 0);
+  notes[1].addModifier(new Articulation('a-').setPosition(3), 0).addModifier(new Annotation('words'));
+  notes[2].addModifier(new Articulation('a-').setPosition(3), 0);
+  notes[3].addModifier(new Articulation('a-').setPosition(3), 0);
+  notes[2].addModifier(new Articulation('a>').setPosition(3), 0);
+  notes[3].addModifier(new Articulation('a>').setPosition(3), 0);
+  notes[3].addModifier(new Articulation('a@a').setPosition(3), 0);
+
+  const voice = f.Voice().setStrict(false).addTickables(notes);
 
+  new Formatter().joinVoices([voice]).formatToStave([voice], stave);
+
+  f.draw();
+
+  ok(true, 'GraceNoteBasic');
+}
```

Comment:
`beamNotes()` is not required I also think that leaving the modifiers close to the notes makes it easier to understand. This is how I left it:
```js
function graceNoteModifiers(options: TestOptions): void {
  const f = VexFlowTests.makeFactory(options, 700, 130);
  const stave = f.Stave({ x: 10, y: 10, width: 650 });

  const gracenotes = [{ keys: ['b/4'], duration: '8', slash: false }].map(f.GraceNote.bind(f));

  const notes = [
    f
      .StaveNote({ keys: ['c/5'], duration: '4', auto_stem: true })
      .addModifier(f.GraceNoteGroup({ notes: gracenotes }), 0),
    f
      .StaveNote({ keys: ['c/5'], duration: '4', auto_stem: true })
      .addModifier(f.GraceNoteGroup({ notes: gracenotes }), 0)
      .addModifier(new Articulation('a-').setPosition(3), 0),
    f
      .StaveNote({ keys: ['c/5'], duration: '4', auto_stem: true })
      .addModifier(f.GraceNoteGroup({ notes: gracenotes }), 0)
      .addModifier(new Articulation('a-').setPosition(3), 0)
      .addModifier(new Accidental('#')),
    f
      .StaveNote({ keys: ['c/5'], duration: '4', auto_stem: true })
      .addModifier(f.GraceNoteGroup({ notes: gracenotes }), 0)
      .addModifier(new Articulation('a-').setPosition(3), 0)
      .addModifier(new Annotation('words')),
    f
      .StaveNote({ keys: ['c/5'], duration: '4', auto_stem: true })
      .addModifier(f.GraceNoteGroup({ notes: gracenotes }), 0)
      .addModifier(new Articulation('a-').setPosition(3), 0)
      .addModifier(new Articulation('a>').setPosition(3), 0),
    f
      .StaveNote({ keys: ['c/5'], duration: '4', auto_stem: true })
      .addModifier(f.GraceNoteGroup({ notes: gracenotes }), 0)
      .addModifier(new Articulation('a-').setPosition(3), 0)
      .addModifier(new Articulation('a>').setPosition(3), 0)
      .addModifier(new Articulation('a@a').setPosition(3), 0),
  ];

  const voice = f.Voice().setStrict(false).addTickables(notes);

  new Formatter().joinVoices([voice]).formatToStave([voice], stave);

  f.draw();

  ok(true, 'GraceNoteModifiers');
}

```

Reply:
This is also why in new and old images above the number of notes is different.